src/main/java/duke/dukeobject/ExpenseList.java

```diff
@@ -3,50 +3,38 @@
 import duke.exception.DukeException;
 import duke.exception.DukeRuntimeException;
 
+import javax.swing.text.View;
 import java.io.File;
 import java.io.IOException;
 import java.math.BigDecimal;
 import java.time.LocalDateTime;
 import java.util.ArrayList;
 import java.util.Comparator;
 import java.util.List;
+import java.util.stream.Collectors;
 
 public class ExpenseList extends DukeList<Expense> {
 
-    private Comparator<Expense> compareCriteria;
-
-    /**
-     * Defines the criteria of sort comparision as amount.
-     * The {@code Expense} with more amount is regarded as the larger.
-     */
-    private static class SortByAmount implements Comparator<Expense> {
-        public int compare(Expense e1, Expense e2) {
-            return e1.getAmount().compareTo(e2.getAmount());
-        }
+    private static enum SortCriteria {
```

Comment:
enums can have other values, use them to store the comparators instead of writing switch case